### ISSUE TITLE
TypeScript named tuple members should show up in intellisense

### ISSUE THREAD

**Issue by Tyriar** [microsoft/TypeScript#42206]
*TS Template added by @mjbvz*

**TypeScript Version**: 4.2.0-dev.20210104

**Search Terms**
- completionEntryDetails
- suggest / suggestion
- complete / completions

---

For the following TS:

```ts
const a: [x: number, y: number] = [1, 1];
```

This is good:

![image](https://user-images.githubusercontent.com/2193314/103412721-5ebc2e80-4b2b-11eb-82ae-7dab94196bd5.png)

This could be improved by indicating the label for that tuple member on the right side:

![image](https://user-images.githubusercontent.com/2193314/103412745-7b586680-4b2b-11eb-86ee-633a31937544.png)

**Comment by mjbvz** [comment]
Looks like there is actually an exception being thrown here:

```
[Trace  - 20:32:35.228] <semantic> Sending request: completionEntryDetails (24). Response expected: yes. Current queue length: 0
Arguments: {
    "file": "/Users/matb/projects/san/x.ts",
    "line": 2,
    "offset": 6,
    "entryNames": [
        "0"
    ]
}
[Trace  - 20:32:35.234] <semantic> Response received: completionEntryDetails (24). Request took 6 ms. Success: false . Message: Error processing request. Cannot read property 'name' of undefined
TypeError: Cannot read property 'name' of undefined
    at /Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:147561:96
    at Object.forEachUnique (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:116062:34)
    at getDocumentationComment (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:147560:16)
    at SymbolObject.getDocumentationComment (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:147352:49)
    at SymbolObject.getContextualDocumentationComment (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:147375:33)
    at Object.getSymbolDisplayPartsDocumentationAndSymbolKind (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:130237:40)
    at /Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:119357:41
    at Object.runWithCancellationToken (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:44040:28)
    at createCompletionDetailsForSymbol (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:119356:30)
    at Object.getCompletionEntryDetails (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:119338:28)
    at Proxy.getCompletionEntryDetails (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:148356:35)
    at Proxy.<anonymous> (/Users/matb/.vscode-insiders/extensions/bierner.lit-html-1.11.1/node_modules/typescript-template-language-service-decorator/lib/template-language-service-decorator.js:96:24)
    at Proxy.<anonymous> (/Users/matb/.vscode-insiders/extensions/bierner.lit-html-1.11.1/node_modules/typescript-template-language-service-decorator/lib/template-language-service-decorator.js:96:24)
    at /Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:159447:57
    at Object.mapDefined (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:612:30)
    at IOSession.Session.getCompletionEntryDetails (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:159445:33)
    at Session.handlers.ts.Map.ts.getEntries._a.<computed> (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:158306:61)
    at /Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:160051:88
    at IOSession.Session.executeWithRequestId (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:160042:28)
    at IOSession.Session.executeCommand (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:160051:33)
    at IOSession.Session.onMessage (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:160077:35)
    at Interface.<anonymous> (/Users/matb/projects/san/node_modules/typescript/lib/tsserver.js:162729:31)
    at Interface.emit (events.js:315:20)
    at Interface._onLine (readline.js:329:10)
    at Interface._normalWrite (readline.js:474:12)
    at Socket.ondata (readline.js:186:10)
    at Socket.emit (events.js:315:20)
    at addChunk (_stream_readable.js:295:12)
    at readableAddChunk (_stream_readable.js:271:9)
    at Socket.Readable.push (_stream_readable.js:212:10)
    at Pipe.onStreamRead (internal/stream_base_commons.js:186:23)
```

**Comment by ZakMiller** [comment]
This looks to be fixed maybe because of #42278 

Screenshot from 4.2.3
![Screen Shot 2021-03-30 at 4 10 01 PM](https://user-images.githubusercontent.com/18072671/113050114-7d55b400-9172-11eb-8ecf-62946810c0a3.png)

**Comment by Tyriar** [comment]
Yes this works now
